src/app.js

```diff
@@ -1,3 +1,62 @@
 'use strict';
+/* eslint-disable no-console */
 
-// Write your code here
+const readline = require('readline');
+const { checkIsValidUserInput } = require('./modules/checkIsValidUserInput');
+const { generateRandomNumber } = require('./modules/generateRandomNumber');
+const { getBullsAndCows } = require('./modules/getBullsAndCows');
+
+const terminal = readline.createInterface(process.stdin, process.stdout);
+
+const guessNumber = generateRandomNumber();
+let count = 0;
+
+function playGame(question = '') {
+  terminal.question(question, (userInput) => {
+    if (!checkIsValidUserInput(userInput)) {
+      console.log(
+        'It is not valid number! Input the 4 dight number, without duplicates!',
+      );
+      playGame('Try again!');
+    } else {
+      count++;
+
+      const result = getBullsAndCows(userInput, guessNumber);
+
+      console.log(result, '(Try', count, ')');
+
+      if (result.bulls === 4) {
+        console.log(
+          'Congratulation!!! You win!!! Number is...  ',
+          guessNumber,
+          ` [From ${count} try!]`,
+        );
+        terminal.close();
+      } else {
+        const { bulls, cows } = result;
+
+        if (bulls === 0 && cows === 0) {
```

Comment:
Recommendation: The conditions on lines 38, 42, and 46 are a bit confusing. It would be clearer to use a switch statement or an if-else-if chain instead. This would also allow you to remove the redundant conditions on lines 47-51.